.github/workflows/branch_preparation.yml

```diff
@@ -0,0 +1,131 @@
+name: Branch Preparation
+
+on:
+  push:
+    branches:
+      - 'release/**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+**'
+
+jobs:
+  prepare_release_branch_after_release:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: cachix/install-nix-action@v22
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./shell.nix
+
+      - name: Test the chart version updater script
+        run: |
+            nix-shell --pure --run "./scripts/test-update-chart-version.sh" ./shell.nix
+
+      - name: Modify the chart version based on the tag
+        run: |
+          tag=${{ github.ref_name }}
+          echo "BASE=$(nix-shell --pure --run "./scripts/update-chart-version.sh --tag $tag" ./shell.nix)" >> $GITHUB_ENV
+
+      - name: Create Pull Request to release
+        if:  ${{ env.BASE }}
+        id: cpr
```

Comment:
Are we missing the automated merge here?

Reply:
No, it's intentional, it would be good to have a glance before merging?

Reply:
Up to you I guess, for mayastor we automated it - still goes via the PR checks and all

Reply:
Yeah, I saw that I can add it, but it would needs the secrets and stuffs to be configured for this repo I think.